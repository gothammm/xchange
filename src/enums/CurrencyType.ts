enum CurrencyType {
  USD = 'USD',
  EUR = 'EUR',
  GBP = 'GBP',
}

namespace CurrencyType {
  export const getSymbol = (currency: CurrencyType) =>
    (0)
      .toLocaleString('en-US', {
        style: 'currency',
        currency: currency,
        minimumFractionDigits: 0,
        maximumFractionDigits: 0,
      })
      .replace(/\d/g, '')
      .trim();
}

export default CurrencyType;
