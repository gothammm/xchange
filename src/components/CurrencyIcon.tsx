import React from 'react';
import CurrencyType from '../enums/CurrencyType';
import { DollarOutlined, EuroOutlined, PoundOutlined, QuestionCircleOutlined } from '@ant-design/icons';

const CurrencyIcon: React.FC<{ currency: CurrencyType }> = ({ currency }) => {
  switch (currency) {
    case CurrencyType.USD:
      return <DollarOutlined />;
    case CurrencyType.EUR:
      return <EuroOutlined />;
    case CurrencyType.GBP:
      return <PoundOutlined />;
    default:
      return <QuestionCircleOutlined />;
  }
};

export default CurrencyIcon;
