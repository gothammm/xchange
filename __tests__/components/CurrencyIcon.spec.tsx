import React from 'react';
import CurrencyIcon from '../../src/components/CurrencyIcon';
import CurrencyType from '../../src/enums/CurrencyType';
import { render } from '@testing-library/react';

describe(`CurrencyIcon`, () => {
  test(`to match snapshot`, () => {
    const component = render(<CurrencyIcon currency={CurrencyType.USD} />);
    expect(component.asFragment()).toMatchSnapshot();

    component.rerender(<CurrencyIcon currency={CurrencyType.EUR} />);
    expect(component.asFragment()).toMatchSnapshot();

    component.rerender(<CurrencyIcon currency={CurrencyType.GBP} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
