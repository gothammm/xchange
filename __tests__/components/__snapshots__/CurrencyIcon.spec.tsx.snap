// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyIcon to match snapshot 1`] = `
<DocumentFragment>
  <span
    aria-label="dollar"
    class="anticon anticon-dollar"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="dollar"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`CurrencyIcon to match snapshot 2`] = `
<DocumentFragment>
  <span
    aria-label="euro"
    class="anticon anticon-euro"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="euro"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm117.7-588.6c-15.9-3.5-34.4-5.4-55.3-5.4-106.7 0-178.9 55.7-198.6 149.9H344c-4.4 0-8 3.6-8 8v27.2c0 4.4 3.6 8 8 8h26.4c-.3 4.1-.3 8.4-.3 12.8v36.9H344c-4.4 0-8 3.6-8 8V568c0 4.4 3.6 8 8 8h30.2c17.2 99.2 90.4 158 200.2 158 20.9 0 39.4-1.7 55.3-5.1 3.7-.8 6.4-4 6.4-7.8v-42.8c0-5-4.6-8.8-9.5-7.8-14.7 2.8-31.9 4.1-51.8 4.1-68.5 0-114.5-36.6-129.8-98.6h130.6c4.4 0 8-3.6 8-8v-27.2c0-4.4-3.6-8-8-8H439.2v-36c0-4.7 0-9.4.3-13.8h135.9c4.4 0 8-3.6 8-8v-27.2c0-4.4-3.6-8-8-8H447.1c17.2-56.9 62.3-90.4 127.6-90.4 19.9 0 37.1 1.5 51.7 4.4a8 8 0 009.6-7.8v-42.8c0-3.8-2.6-7-6.3-7.8z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`CurrencyIcon to match snapshot 3`] = `
<DocumentFragment>
  <span
    aria-label="pound"
    class="anticon anticon-pound"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="pound"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm138-209.8H469.8v-4.7c27.4-17.2 43.9-50.4 43.9-91.1 0-14.1-2.2-27.9-5.3-41H607c4.4 0 8-3.6 8-8v-30c0-4.4-3.6-8-8-8H495c-7.2-22.6-13.4-45.7-13.4-70.5 0-43.5 34-70.2 87.3-70.2 21.5 0 42.5 4.1 60.4 10.5 5.2 1.9 10.6-2 10.6-7.6v-39.5c0-3.3-2.1-6.3-5.2-7.5-18.8-7.2-43.8-12.7-70.3-12.7-92.9 0-151.5 44.5-151.5 120.3 0 26.3 6.9 52 14.6 77.1H374c-4.4 0-8 3.6-8 8v30c0 4.4 3.6 8 8 8h67.1c3.4 14.7 5.9 29.4 5.9 44.2 0 45.2-28.8 83.3-72.8 94.2-3.6.9-6.1 4.1-6.1 7.8V722c0 4.4 3.6 8 8 8H650c4.4 0 8-3.6 8-8v-39.8c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
